tests/security/grub_auth/grub_authentication.pm

```diff
@@ -0,0 +1,27 @@
+# Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# Summary: enter username and password when grub is protected
+#
+# Maintainer: QE Security <none@suse.de>
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use base 'consoletest';
+
+my $user = 'root';
```

Comment:
you could use `$testapi::username` and `$testapi::password`

Reply:
seems standard test username is not allowed to unlock 'grub', we need root here. Updated also the test reliability adding a needle to ensure the correct screen is displayed